packages/db/prisma/extensions/base.ts

```diff
@@ -101,19 +101,32 @@ export const baseExtension = Prisma.defineExtension((prisma) =>
         },
       },
       blob: {
-        async filterNewBlobs(blobs: RawBlob[]) {
-          const existingBlobVersionedHashes = (
+        async filterNewBlobs(
+          blobs: RawBlob[]
+        ): Promise<Omit<RawBlob, "index">[]> {
+          const dbBlobVersionedHashes = (
             await prisma.blob.findMany({
               select: { versionedHash: true },
               where: {
                 versionedHash: { in: blobs.map((blob) => blob.versionedHash) },
               },
             })
           ).map((b) => b.versionedHash);
-
-          return blobs.filter(
-            (b) => !existingBlobVersionedHashes.includes(b.versionedHash)
-          );
+          const newBlobVersionedHashes = Array.from(
+            new Set(blobs.map((b) => b.versionedHash))
+          ).filter((hash) => !dbBlobVersionedHashes.includes(hash));
+
+          return newBlobVersionedHashes.map((versionedHash) => {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const b = blobs.find((b) => b.versionedHash === versionedHash)!;
```

Comment:
I think we are repeating the same logic twice here. Instead of having a `newBlobVersionedHashes` we can keep all the `newBlobs` in the first filter and return that. What do you think?

Reply:
We pass the blob's versioned hashes to be indexed through a set to remove any possible duplicates as a block can contain the same blob multiple times. Then we filter out the blobs that are already stored in the db.

Next, we iterate over these filtered versioned hashes to build the full blob again.

We can use the exclamation mark operator without any issues, as the resulting filtered blobs are a subset of the original blobs received

Reply:
Perhaps we could add a comment here as well, to provide some clarity on this logic.

Reply:
Cool, thank you for explain it. I did not factored the fact that we need the `Set` includes method to make filter easier.